contrib/globalsign/mgo/mgo.go

```diff
@@ -0,0 +1,395 @@
+package mgo
+
+import (
+	"context"
+
+	"github.com/globalsign/mgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+type mongoConfig struct {
+	serviceName string
+}
+
+// MongoOption represents an option that can be passed to Dial
+type MongoOption func(*mongoConfig)
+
+func defaults(cfg *mongoConfig) {
+	cfg.serviceName = "mongodb"
+}
+
+// WithServiceName sets the service name for a given MongoDB context.
+func WithServiceName(name string) MongoOption {
+	return func(cfg *mongoConfig) {
+		cfg.serviceName = name
+	}
+}
+
+// Dial opens a connection to a MongoDB server and ties it to the
+// given context for tracing MongoDB calls.
+func Dial(ctx context.Context, url string, opts ...MongoOption) (*WrapSession, error) {
+	session, err := mgo.Dial(url)
+	s := &WrapSession{
+		Session: session,
+		ctx:     ctx,
+	}
+
+	defaults(&s.cfg)
+	for _, fn := range opts {
+		fn(&s.cfg)
+	}
+
+	return s, err
+}
+
+// WrapSession wraps a Session with the Context and Config information
+// needed to support tracing.
+type WrapSession struct {
+	*mgo.Session
+
+	ctx context.Context
+	cfg mongoConfig
+}
+
+func newChildSpanFromContext(ctx context.Context, config mongoConfig, resource string, op string) ddtrace.Span {
+	name := op
+	span, _ := tracer.StartSpanFromContext(
+		ctx,
+		name,
+		tracer.SpanType("mongodb"),
+		tracer.ServiceName(config.serviceName))
+	span.SetTag("resource.name", resource)
```

Comment:
nitpick: you can set this in the `StartSpanFromContext` with `tracer.ResourceName`

Reply:
Np.